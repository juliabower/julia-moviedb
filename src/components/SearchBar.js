import React from "react";

const SearchBar = () => {
  return (
    <div className="row">
        <div class="col-lg-12">
          <div class="input-group">
            <input type="text" className="form-control" placeholder="e.g. Harry Potter" />
            <span className="input-group-btn">
              <button className="btn btn-default btn-go" type="button">Go!</button>
            </span>
          </div>
        </div>
      </div>
  );
}

export default SearchBar;