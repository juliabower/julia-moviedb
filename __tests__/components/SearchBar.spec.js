import React from "react";
import { shallow } from "enzyme";
import SearchBar from "../../src/components/SearchBar";

describe('SearchBar', () => {
  let componentUnderTest;
  const mockSearch = jest.fn();

  const getComponent = () => {
    if (!componentUnderTest) {
      componentUnderTest = shallow(
        <SearchBar search={mockSearch} />
      );
    }
    return componentUnderTest;
  };

  beforeEach(() => {
    componentUnderTest = undefined;
  })

  describe('SearchBar tests', () => {
    let component;

    beforeEach(() => {
      component = getComponent();
    })
    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });

    it('should call search onclick', () => {
      component.find('.btn-go').simulate('click', { preventDefault() {} });
      expect(mockSearch).toHaveBeenCalled();
    });

    it('should update state correctly', () => {
      expect(component.state()).toEqual({search: ''});
      component.find('input').simulate('change', {target: {value: 'test'}});
      expect(component.state()).toEqual({search: 'test'});
    });
  });
});