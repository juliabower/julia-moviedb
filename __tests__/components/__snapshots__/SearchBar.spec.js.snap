// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar tests should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    search={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="e.g. Harry Potter"
              type="text"
            />
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default btn-go"
                onClick={[Function]}
                type="button"
              >
                Go
              </button>
            </span>
          </div>
        </div>
      </div>,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-lg-12"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="e.g. Harry Potter"
              type="text"
            />
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default btn-go"
                onClick={[Function]}
                type="button"
              >
                Go
              </button>
            </span>
          </div>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="e.g. Harry Potter"
              type="text"
            />
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default btn-go"
                onClick={[Function]}
                type="button"
              >
                Go
              </button>
            </span>
          </div>,
          "className": "col-lg-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="e.g. Harry Potter"
                type="text"
              />,
              <span
                className="input-group-btn"
              >
                <button
                  className="btn btn-default btn-go"
                  onClick={[Function]}
                  type="button"
                >
                  Go
                </button>
              </span>,
            ],
            "className": "input-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "onChange": [Function],
                "placeholder": "e.g. Harry Potter",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-default btn-go"
                  onClick={[Function]}
                  type="button"
                >
                  Go
                </button>,
                "className": "input-group-btn",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Go",
                  "className": "btn btn-default btn-go",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Go",
                "type": "button",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-lg-12"
          >
            <div
              className="input-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="e.g. Harry Potter"
                type="text"
              />
              <span
                className="input-group-btn"
              >
                <button
                  className="btn btn-default btn-go"
                  onClick={[Function]}
                  type="button"
                >
                  Go
                </button>
              </span>
            </div>
          </div>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-lg-12"
          >
            <div
              className="input-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="e.g. Harry Potter"
                type="text"
              />
              <span
                className="input-group-btn"
              >
                <button
                  className="btn btn-default btn-go"
                  onClick={[Function]}
                  type="button"
                >
                  Go
                </button>
              </span>
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="e.g. Harry Potter"
                type="text"
              />
              <span
                className="input-group-btn"
              >
                <button
                  className="btn btn-default btn-go"
                  onClick={[Function]}
                  type="button"
                >
                  Go
                </button>
              </span>
            </div>,
            "className": "col-lg-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="e.g. Harry Potter"
                  type="text"
                />,
                <span
                  className="input-group-btn"
                >
                  <button
                    className="btn btn-default btn-go"
                    onClick={[Function]}
                    type="button"
                  >
                    Go
                  </button>
                </span>,
              ],
              "className": "input-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "onChange": [Function],
                  "placeholder": "e.g. Harry Potter",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-default btn-go"
                    onClick={[Function]}
                    type="button"
                  >
                    Go
                  </button>,
                  "className": "input-group-btn",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Go",
                    "className": "btn btn-default btn-go",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Go",
                  "type": "button",
                },
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
